controller_list:
  - name: "blue_arm_controller"
    action_ns: follow_joint_trajectory
    type: FollowJointTrajectory
    default: true
    joints:
      - joint_1
      - joint_2
      - joint_3
      - joint_4
      - joint_5
      - joint_6
      - joint_7
  - name: "joint_state_controller"
    type: joint_state_controller/JointStateController
    publish_rate: 25

# blue_arm:
joint_state_controller:
  type: joint_state_controller/JointStateController
  publish_rate: 25

blue_arm_controller:
  # type: position_controllers/JointTrajectoryController
  type: velocity_controllers/JointTrajectoryController
  joints:
    - joint_1
    - joint_2
    - joint_3
    - joint_4
    - joint_5
    - joint_6
    - joint_7
  constraints:
      goal_time: 1
      stopped_velocity_tolerance: 0.05
      joint_1: {trajectory: 0.1, goal: 0.1}
      joint_2: {trajectory: 0.1, goal: 0.1}
      joint_3: {trajectory: 0.1, goal: 0.1}
      joint_4: {trajectory: 0.1, goal: 0.1}
      joint_5: {trajectory: 0.1, goal: 0.1}
      joint_6: {trajectory: 0.1, goal: 0.1}
      joint_7: {trajectory: 0.1, goal: 0.1}
  stop_trajectory_duration: 0.5
  state_publish_rate:  25
  action_monitor_rate: 12
  gains:
    joint_1:
      p: 2
      d: 0
      i: 0
    joint_2:
      p: 2
      d: 0
      i: 0
    joint_3:
      p: 2
      d: 0
      i: 0
    joint_4:
      p: 2
      d: 0
      i: 0
    joint_5:
      p: 2
      d: 0
      i: 0
    joint_6:
      p: 2
      d: 0
      i: 0
    joint_7:
      p: 2
      d: 0
      i: 0

joint_limits:
  joint_1:
    has_position_limits: true
    min_position: -3.1415926
    max_position: 3.1415926
    has_velocity_limits: true
    max_velocity: 3.29
    has_acceleration_limits: true
    max_acceleration: 6.58
  joint_2:
    has_position_limits: true
    min_position: 0.0
    max_position: 3.1415926
    has_velocity_limits: true
    max_velocity: 3.29
    has_acceleration_limits: true
    max_acceleration: 6.58
  joint_3:
    has_position_limits: true
    min_position: -3.1415926
    max_position: 3.1415926
    has_velocity_limits: true
    max_velocity: 2.39
    has_acceleration_limits: true
    max_acceleration: 4.78
  joint_4:
    has_position_limits: true
    min_position: -3.1415926
    max_position: 0.0
    has_velocity_limits: true
    max_velocity: 2.39
    has_acceleration_limits: true
    max_acceleration: 4.78
  joint_5:
    has_position_limits: true
    min_position: -3.1415926
    max_position: 3.1415926
    has_velocity_limits: true
    max_velocity: 2.39
    has_acceleration_limits: true
    max_acceleration: 4.78
  joint_6:
    has_position_limits: true
    min_position: -1.7
    max_position: 1.7
    has_velocity_limits: true
    max_velocity: 2.77
    has_acceleration_limits: true
    max_acceleration: 5.54
  joint_7:
    has_position_limits: true
    min_position: -3.1415926
    max_position: 3.1415926
    has_velocity_limits: true
    max_velocity: 2.77
    has_acceleration_limits: true
    max_acceleration: 5.54